### PR TITLE
Implement U016: Remove six compatibility code

### PR DESCRIPTION
**Pull request by martinlehoux**
- #827 
- There are still a few things to fix, but I've been holding onto this one for too long

### CODE REVIEW

**Comment by martinlehoux** [comment]
@charliermarsh FYI

File: resources/test/fixtures/pyupgrade/U016.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class C(six.Iterator): pass # class C: pass
+
+class C(six.with_metaclass(M, B)): pass # class C(B, metaclass=M): pass
+
+# class C(B, metaclass=M): pass
+@six.add_metaclass(M)
+class C(B): pass
+
+
+next(six.iteritems(dct)) # next(iter(dct.items()))
```
Comment (charliermarsh) [inline resources/test/fixtures/pyupgrade/U016.py:80]: We should probably correct this one to `next(iter(dct.items()))` (right now it's `next(dct.items())`) before merging.

File: resources/test/fixtures/pyupgrade/U016.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+@six.python_2_unicode_compatible # Remove
+
+# Rewrite classes
+class C(six.Iterator): pass # class C: pass
+
+class C(six.with_metaclass(M, B)): pass # class C(B, metaclass=M): pass
+
+# class C(B, metaclass=M): pass
+@six.add_metaclass(M)
+class C(B): pass
```
Comment (charliermarsh) [inline resources/test/fixtures/pyupgrade/U016.py:77]: If you want to punt on the class conversions before merging, that's fine.

File: resources/test/fixtures/pyupgrade/U016.py (hunk 1 above)
Comment (charliermarsh) [reply resources/test/fixtures/pyupgrade/U016.py:80]: Are you up for that @martinlehoux?

File: src/pyupgrade/plugins/remove_six_compat.rs [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        .map_or(false, |module_name| *module_name == module)
+}
+
+fn map_name(name: &str, expr: &Expr, patch: bool) -> Option<Check> {
+    let replacement = match name {
+        "text_type" => Some("str"),
+        "binary_type" => Some("bytes"),
+        "class_types" => Some("type"),
+        "string_types" => Some("str"),
+        "integer_types" => Some("int"),
```
[the remark this answers is not in the document]
Comment (martinlehoux) [reply src/pyupgrade/plugins/remove_six_compat.rs:26]: No I guess you're right!

File: resources/test/fixtures/pyupgrade/U016.py (hunk 2 above)
Comment (martinlehoux) [reply resources/test/fixtures/pyupgrade/U016.py:77]: I'll have a try today, I'll let you know if I can address this

File: resources/test/fixtures/pyupgrade/U016.py (hunk 1 above)
Comment (martinlehoux) [reply resources/test/fixtures/pyupgrade/U016.py:80]: Do you have any advice on how I could do that? 

I guess going through the LST is done from higher level to lower level, so if I rewrite `next(six.iteritems(dct))` to `next(iter(dct.items()))`, then there won't be a "lower level" issue to fix (what is getting fixed today)

File: src/pyupgrade/plugins/remove_six_compat.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (martinlehoux) [reply src/pyupgrade/plugins/remove_six_compat.rs:26]: Okay I dodn't see you made this change when I answered. I must say I don't have any background on `six` usage, but I took the examples from pyupgrade as test, so Iguess yes, it was intentional to have single element tuples.

[the remark this answers is not in the document]
Comment (martinlehoux) [reply src/pyupgrade/plugins/remove_six_compat.rs:26]:
```
>>> import six
>>> six.class_types
(<class 'type'>,)
```

File: resources/test/fixtures/pyupgrade/U016.py (hunk 1 above)
Comment (martinlehoux) [reply resources/test/fixtures/pyupgrade/U016.py]: seems to work

File: resources/test/fixtures/pyupgrade/U016.py (hunk 2 above)
Comment (martinlehoux) [reply resources/test/fixtures/pyupgrade/U016.py]: Sorry I didn't take the time, let's merge this one before and then I'll get on classes

**Comment by martinlehoux** [comment]
@charliermarsh Looks good to me now

**Pull request merged**